src/nuketesting/_cli/runner.py

```diff
@@ -0,0 +1,162 @@
+        """
+        if not executable.exists():
+            msg = f"Provided nuke path '{executable}' does not exist."
+            raise RunnerException(msg)
+
+    def _find_nuke_python_package(self) -> Path | None:
+        """Find the Nuke Python package based on the executable directory.
+
+        Raises:
+            RunnerException: when run on MacOS
+
+        Returns:
+            path to site packages of Nuke.
+        """
+        if platform.system().lower() == "darwin":
+            msg = "On MacOS the tests can only run in terminal mode."
+            raise RunnerException(msg)
+        python_version = sys.version_info
+        packages_path = (
+            self._nuke_executable.parent
+            / "lib"
+            / f"python{python_version.major}.{python_version.minor}"
+            / "site-packages"
+        )
+        if not packages_path.is_dir():
+            msg = (
+                "The current python version does not match the Nuke version. "
+                f"This is Python: {python_version.major}.{python_version.minor}. "
+                "Please run in terminal mode instead."
+            )
+            raise RunnerException(msg)
+        return packages_path
+
+    def execute_tests(self, test_path: str | Path) -> int:
+        """Run the testrunner with provided arguments.
+
+        Args:
+            test_path: filepath to the tests. Can be relative to the current working directory.
+                       Individual tests can be executed with the file.py::TestClass::test_function
+                       syntax. For more details consult the pytest documentation.
+        """
+        return self._execute_in_nuke(test_path) if self._run_in_terminal_mode else self._execute_native(test_path)
+
+    def _get_packages_directory(self) -> Path:
+        """Get the PATH to the packages locations necessary for running tests."""
+        packages_directory = Path(pytest.__file__).parent.parent
+        testrunner_directory = Path(nuketesting.__file__).parent.parent
+        return f"{packages_directory!s}:{testrunner_directory!s}"
+
+    def _execute_in_nuke(self, test_path: str | Path) -> int:
+        """Execute the tests using the Nuke interpreter.
+
+        Args:
+            test_path: path to tests
+
+        Returns:
+            int: exitcode of tests
+        """
+        packages_directory = self._get_packages_directory()
+        try:
+            arguments = [
+                str(self._nuke_executable),
+                *self._executable_args,
+                "-t",
+                f"'{self.TEST_SCRIPT!s}'",
+                f"--packages_directory '{packages_directory!s}'",
+                f"--test_dir '{test_path!s}'",
+            ]
+            if self._pytest_args:
+                pytest_args = [f'--pytest_arg "{arg}"' for arg in self._pytest_args]
+                arguments.extend(pytest_args)
+            arguments = " ".join(arguments)
+            subprocess.call(arguments, shell=True)
```

Comment:
I added it on purpose as we want to invoke a shell interface to get the outputs of the command in real time, like we do when using pytest

Reply:
Ok, but this can be archived by pipeing the stdout and stderr without a shell. The shell way is also very OS dependant because you never know which shell will be used (bash, Powershell, dos etc)